provision.sh

```diff
@@ -10,6 +16,19 @@ package() {
     sudo apt-get -y install "$@"
 }
 
+git_clone() {
+    # Clone git repo into directory, clearing directory if it exists
+    # Usage: git_clone URL DIR [GIT_CLONE OPTIONS...]
+
+    local url="$1"
+    shift
+    local dir="$1"
```

Comment:
If the URL just ends in `.git`, then yes. The `git-clone(1)` man page page states:
> The name of a new directory to clone into. The "humanish" part of the source repository is used if no directory is explicitly given (repo for /path/to/repo.git and foo for host.xz:foo/.git). Cloning into an existing directory is only allowed if the directory is empty.

I thought it would be better to explicitly include the directory name instead of trying to parse the clone URL and handle several cases.